pkg/ec2helper/ec2helper.go

```diff
@@ -352,76 +353,137 @@ func (h *EC2Helper) getInstanceTypes(input *ec2.DescribeInstanceTypesInput) ([]*
 // Define all OS and corresponding AMI name formats
 var osDescs = map[string]map[string]string{
 	"Amazon Linux": {
-		"ebs":            "amzn-ami-hvm-????.??.?.????????.?-*-gp2",
-		"instance-store": "amzn-ami-hvm-????.??.?.????????.?-*-s3",
+		"ebs":            "amzn-ami-hvm-*",
+		"instance-store": "amzn-ami-hvm-*",
 	},
 	"Amazon Linux 2": {
-		"ebs": "amzn2-ami-hvm-2.?.????????.?-*-gp2",
+		"ebs": "amzn2-ami-hvm-*",
 	},
 	"Red Hat": {
-		"ebs": "RHEL-?.?.?_HVM-????????-*-?-Hourly2-GP2",
+		"ebs": "RHEL-9*",
 	},
 	"SUSE Linux": {
-		"ebs": "suse-sles-??-sp?-v????????-hvm-ssd-*",
+		"ebs": "suse-sles-*",
 	},
-	// Ubuntu 18.04 LTS
 	"Ubuntu": {
-		"ebs":            "ubuntu/images/hvm-ssd/ubuntu-bionic-18.04-*-server-????????",
-		"instance-store": "ubuntu/images/hvm-instance/ubuntu-bionic-18.04-*-server-????????",
+		"ebs":            "ubuntu/images/*",
+		"instance-store": "ubuntu/images/*",
 	},
-	// 64 bit Microsoft Windows Server with Desktop Experience Locale English AMI
 	"Windows": {
-		"ebs": "Windows_Server-????-English-Full-Base-????.??.??",
+		"ebs": "Windows_Server-*-English-Full-Base*",
+	},
+}
+
+// Define all OS and corresponding AMI public parameters path in Parameter Store
+var osSsmPath = map[string]map[string]string{
+	"Amazon Linux": {
+		"ebs":            "/aws/service/ami-amazon-linux-latest",
+		"instance-store": "/aws/service/ami-amazon-linux-latest",
+	},
+	"Amazon Linux 2": {
+		"ebs": "/aws/service/ami-amazon-linux-latest",
+	},
+	"Red Hat": {
+		"ebs": "",
+	},
+	"SUSE Linux": {
+		"ebs": "/aws/service/suse/sles",
+	},
+	"Ubuntu": {
+		"ebs":            "/aws/service/canonical/ubuntu/server/24.04/stable/current",
+		"instance-store": "/aws/service/canonical/ubuntu/server/24.04/stable/current",
+	},
+	"Windows": {
+		"ebs": "/aws/service/ami-windows-latest",
 	},
 }
 
 // Get the appropriate input for describing images
-func getDescribeImagesInputs(rootDeviceType string, architectures []*string) *map[string]ec2.DescribeImagesInput {
+func (h *EC2Helper) getDescribeImagesInputs(rootDeviceType string, architectures []*string) *map[string]ec2.DescribeImagesInput {
+	ssmClient := ssm.New(h.Sess)
+
 	// Construct all the inputs
 	imageInputs := map[string]ec2.DescribeImagesInput{}
 	for osName, rootDeviceTypes := range osDescs {
 
 		// Only add inputs if the corresponding root device type is applicable for the specified os
 		desc, found := rootDeviceTypes[rootDeviceType]
-		if found {
-			imageInputs[osName] = ec2.DescribeImagesInput{
-				Filters: []*ec2.Filter{
-					{
-						Name: aws.String("name"),
-						Values: []*string{
-							aws.String(desc),
-						},
-					},
-					{
-						Name: aws.String("state"),
-						Values: []*string{
-							aws.String("available"),
-						},
+		if !found {
+			continue
```

Comment:
continue as in? getting default old images?

Reply:
Here is just refactoring. There is no modification on original logic.